tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -442,18 +467,19 @@ impl<E> Deref for FormattedFields<E> {
 
 // === impl FmtLayer ===
 
-macro_rules! event_from_span {
-    ($event:ident = $id:ident, $span:ident, $($field:ident = $value:expr),*) => {
+macro_rules! with_event_from_span {
+    ($id:ident, $span:ident, $($field:literal = $value:expr),*, |$event:ident| $code:tt) => {
```

Comment:
nit: I think we could use a `block` fragment specifier here rather than `tt`:
```suggestion
    ($id:ident, $span:ident, $($field:literal = $value:expr),*, |$event:ident| $code:block) => {
```